src/galileo/base.py

```diff
@@ -9,13 +10,20 @@ class BaseClientModel:
     Used by the model classes to lazy load the API client.
 
     Args:
-        client (Optional[GalileoApiClient], optional): The client to use. Defaults to None.
+        config (Optional[GalileoConfig], optional): The config to use. Defaults to None.
```

Comment:
Also curious why we need this class... This just looks like a silly wrapper on top of the config class?

Reply:
That's exactly what it is : ) I subclassed it based on our brief conversation in slack, but I'm not actually using it to maintain any state.. it was more of a, "for the future", kinda thing.

I can remove and just use the core Config class directly.. or I can put some state in there, but I haven't looked properly into how the other repos are using their custom config classes to maintain/use the state.

<img width="1045" height="332" alt="Screenshot 2025-07-28 at 7 49 24 PM" src="https://github.com/user-attachments/assets/af90b7c3-8665-4d2c-bb73-be5bd8149e23" />

Reply:
I was actually suggesting us using `GalileoConfig` directly / sub-classing it in here (which exists in the other file) but not using that in the init for this... Right now, we have a 3-layer thing: `BaseClient` => `Config` => `APIClient`, which seems a few too many abstractions.

Reply:
Oh I see.. you mean `BaseClientModel` is the silly wrapper.. yea, I tried changing as little as possible to be a bit more expeditious, which left a bit of cruft in there for sure. 

I'll circle back on this one and add a ticket to address this in the near future.  

Slightly under the weather since yesterday so I'm a bit cloudy and the math just ain't math'ing in my head at the mo.

Reply:
Sounds good, thanks Jonathan.

Reply:
Wlc